### PR TITLE
Fix stress issues with a test

### LINKED ISSUE #2790: SqlServerStorageTests.WriteReadWriteRead100StatesInParallel test failing
Getting the following error when running the functionals lately:
`WriteReadWriteRead100StatesInParallel [0:37.528] Failed: Orleans.Storage.InconsistentStateException: Version conflict (WriteState): ServiceId=00000000-0000-0000-0000-000000000000 ProviderName=System.Data.SqlClient_StorageProvider GrainType=UnitTests.StorageTests.Relational.TestDataSets.GrainTypeGenerator+TestGrainGenericWithGuidKey'1[[System.Double, mscorlib]] GrainId=33900 ETag=.`

@veikkoeeva / @shayhatsor do you know of anything that changed in the relational code that might be causing this?
Full stacktrace below:

```
Orleans.Storage.InconsistentStateException
Version conflict (WriteState): ServiceId=00000000-0000-0000-0000-000000000000 ProviderName=System.Data.SqlClient_StorageProvider GrainType=UnitTests.StorageTests.Relational.TestDataSets.GrainTypeGenerator+TestGrainGenericWithGuidKey`1[[System.Double, mscorlib]] GrainId=33900 ETag=.
   at Orleans.Storage.AdoNetStorageProvider.<WriteStateAsync>d__43.MoveNext() in D:\projects\orleans\src\OrleansSQLUtils\Storage\Provider\AdoNetStorageProvider.cs:line 449
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at UnitTests.StorageTests.Relational.CommonStorageTests.<Store_WriteRead>d__9`1.MoveNext() in D:\projects\orleans\test\TesterInternal\StorageTests\CommonStorageTests.cs:line 157
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at UnitTests.StorageTests.Relational.CommonStorageTests.<>c__DisplayClass5_1.<<PersistenceStorage_WriteReadWriteReadStatesInParallel>b__1>d.MoveNext() in D:\projects\orleans\test\TesterInternal\StorageTests\CommonStorageTests.cs:line 67
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at UnitTests.StorageTests.Relational.CommonStorageTests.<PersistenceStorage_WriteReadWriteReadStatesInParallel>d__5.MoveNext() in D:\projects\orleans\test\TesterInternal\StorageTests\CommonStorageTests.cs:line 79
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at UnitTests.StorageTests.Relational.RelationalStorageTests.<Relational_WriteReadWriteRead100StatesInParallel>d__25.MoveNext() in D:\projects\orleans\test\TesterSQLUtils\StorageTests\Relational\RelationalStorageTests.cs:line 67
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at UnitTests.StorageTests.Relational.SqlServerStorageTests.<WriteReadWriteRead100StatesInParallel>d__4.MoveNext() in D:\projects\orleans\test\TesterSQLUtils\StorageTests\Relational\SqlServerStorageTests.cs:line 50
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 227
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.ExecutionTimer.<AggregateAsync>d__4.MoveNext() in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.ExceptionAggregator.<RunAsync>d__9.MoveNext() in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90


```

### PR DESCRIPTION
**Pull request by jdom**
* Avoids stressing the storage on the INSERT call by limiting parallelization on the first write.
* Avoids starving the thread pool while doing concurrent updates, but still force the store to be stressed a little bit.

Fixes #2790

### CODE REVIEW

File: test/TesterInternal/StorageTests/CommonStorageTests.cs [hunk 1]
```diff
@@ -46,36 +46,48 @@ public CommonStorageTests(IInternalGrainFactory grainFactory, IStorageProvider s
...
+                    //version is updated appropriately.
+                    await Store_WriteRead(grainTypeName, grainData.Item1, grainData.Item2);
+                    var secondVersion = grainData.Item2.ETag;
                     Assert.NotEqual(firstVersion, secondVersion);
-                    Assert.NotEqual(secondVersion, thirdVersion);
-                }
-            }));
+                }));
+
+            int maxNumberOfThreads = 25;
```
Comment (veikkoeeva) [inline test/TesterInternal/StorageTests/CommonStorageTests.cs]: To be a nitpicker: `const int MaxNumberOfThreads = 25`? :)

File: test/TesterInternal/StorageTests/CommonStorageTests.cs [hunk 2]
```diff
@@ -46,36 +46,48 @@ public CommonStorageTests(IInternalGrainFactory grainFactory, IStorageProvider s
...
-                    var secondVersion = grainData.Item2.ETag;
-                    Assert.NotEqual(firstVersion, secondVersion);
-                    await Store_WriteRead(grainTypeName, grainData.Item1, grainData.Item2);
+                    //A sanity checker that the first version really has null as its state. Then it is stored
+                    //to the database and a new version is acquired.
+                    var firstVersion = grainData.Item2.ETag;
+                    Assert.Equal(firstVersion, null);
 
-                    var thirdVersion = grainData.Item2.ETag;
+                    //This loop writes the state consecutive times to the database to make sure its
```
Comment (veikkoeeva) [inline test/TesterInternal/StorageTests/CommonStorageTests.cs]: This comment looks like being a left-over, the update loop is split from the previous _insert first and then update in loop_ task.

File: test/TesterInternal/StorageTests/CommonStorageTests.cs [hunk 3]
```diff
@@ -46,36 +46,48 @@ public CommonStorageTests(IInternalGrainFactory grainFactory, IStorageProvider s
...
-                Assert.Equal(firstVersion, null);
-
-                //This loop writes the state consecutive times to the database to make sure its
-                //version is updated appropriately.
-                await Store_WriteRead(grainTypeName, grainData.Item1, grainData.Item2);
-                for(int k = 0; k < 10; ++k)
+            //Since the version is NULL, storage provider tries to insert this data
+            //as new state. If there is already data with this class, the writing fails
+            //and the storage provider throws. Essentially it means either this range
+            //is ill chosen or the test failed due another problem.
```
Comment (veikkoeeva) [inline test/TesterInternal/StorageTests/CommonStorageTests.cs]: Maybe we could fix my previous errors in the comments in the same go too?
> ... due to another problem...

**Comment by jdom** [comment]
Thanks, @veikkoeeva, fixed them!

**Comment by jdom** [comment]
And verified that it didn't fail in the funtional run in the CI server.

**Comment by veikkoeeva** [comment]
:+1:

**Pull request merged**